src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -773,269 +774,292 @@ static HtmlEntities()
+                    [7364963] = '\x2229', // "cap"
+                    [7370083] = '\x222a', // "cup"
+                    [7630441] = '\x222b', // "int"
+                    [57610315589748] = '\x2234', // "there4"
+                    [7170419] = '\x223c', // "sim"
+                    [1735290723] = '\x2245', // "cong"
+                    [482873013089] = '\x2248', // "asymp"
+                    [25966] = '\x2260', // "ne"
+                    [508575445349] = '\x2261', // "equiv"
+                    [25964] = '\x2264', // "le"
+                    [25959] = '\x2265', // "ge"
+                    [6452595] = '\x2282', // "sub"
+                    [7370099] = '\x2283', // "sup"
+                    [1651864430] = '\x2284', // "nsub"
+                    [1700951411] = '\x2286', // "sube"
+                    [1701868915] = '\x2287', // "supe"
+                    [495891279983] = '\x2295', // "oplus"
+                    [126879464518767] = '\x2297', // "otimes"
+                    [1886545264] = '\x22a5', // "perp"
+                    [1953457267] = '\x22c5', // "sdot"
+                    [465624720236] = '\x2308', // "lceil"
+                    [465624720242] = '\x2309', // "rceil"
+                    [125822936311404] = '\x230a', // "lfloor"
+                    [125822936311410] = '\x230b', // "rfloor"
+                    [1735287148] = '\x2329', // "lang"
+                    [1735287154] = '\x232a', // "rang"
+                    [8023916] = '\x25ca', // "loz"
+                    [126879312998515] = '\x2660', // "spades"
+                    [495573101667] = '\x2663', // "clubs"
+                    [126943972386152] = '\x2665', // "hearts"
+                    [495756339556] = '\x2666', // "diams"
+                    // To compute the key for new entries, use:
+                    // string entity = "diams"; ulong key = 0; for (int i = 0; i < entity.Length; i++) key |= (ulong)entity[i] << 8 * i; Console.WriteLine(key);
                 };
 
-            public static char Lookup(string entity)
+            public static char Lookup(ReadOnlySpan<char> entity)
             {
-                char theChar;
-                s_lookupTable.TryGetValue(entity, out theChar);
-                return theChar;
+                // To avoid an allocation, keys of type "ulong" are used in the lookup table.
+                // Since all entity strings comprise 8 characters or less and are ASCII-only, they "fit" into an ulong (8 bytes).
+                // The ulong key is the reversed single-byte character representation of the actual entity string.
+                if (entity.Length <= 8)
+                {
+                    ulong key = 0;
+                    for (int i = 0; i < entity.Length; i++)
+                    {
+                        if (entity[i] > 0xFF)
+                        {
+                            return default;
+                        }
+
+                        key |= (ulong)entity[i] << 8 * i;
```

Comment:
Shift by variable amount tends to be more expensive than shift by fixed amount. It may be better to do it other way: `key = (key << 8) | entity[i]`